flink-end-to-end-tests/flink-e2e-test-utils/src/main/java/org/apache/flink/streaming/tests/util/s3/S3UtilProgram.java

```diff
@@ -0,0 +1,221 @@
+		handlersMutable.put(Action.numberOfLinesByPrefix, S3UtilProgram::numberOfLinesByPrefix);
+		handlers = Collections.unmodifiableMap(handlersMutable);
+	}
+
+	private static final String countQuery = "select count(*) from s3object";
+
+	public static void main(String[] args) {
+		final ParameterTool params = ParameterTool.fromArgs(args);
+		final Action action = Action.valueOf(params.getRequired("action"));
+		handlers.get(action).accept(params);
+	}
+
+	private static void listByPrefix(ParameterTool params) {
+		final String bucket = params.getRequired("bucket");
+		final String s3prefix = params.getRequired("s3prefix");
+		listByPrefix(bucket, s3prefix).forEach(System.out::println);
+	}
+
+	private static List<String> listByPrefix(final String bucket, final String s3prefix) {
+		return AmazonS3ClientBuilder.defaultClient().listObjects(bucket, s3prefix).getObjectSummaries()
+			.stream().map(S3ObjectSummary::getKey).collect(Collectors.toList());
+	}
+
+	private static void downloadFile(ParameterTool params) {
+		final String bucket = params.getRequired("bucket");
+		final String s3file = params.getRequired("s3file");
+		final String localFile = params.getRequired("localFile");
+		TransferManager tx = TransferManagerBuilder.defaultTransferManager();
+		try {
+			tx.download(bucket, s3file, new File(localFile)).waitForCompletion();
+		} catch (InterruptedException e) {
+			System.out.println("Transfer interrupted");
+		} finally {
+			tx.shutdownNow();
+		}
+	}
+
+	private static void downloadByPrefix(ParameterTool params) {
+		final String bucket = params.getRequired("bucket");
+		final String s3prefix = params.getRequired("s3prefix");
+		final String localFolder = params.getRequired("localFolder");
+		final String s3filePrefix = params.get("s3filePrefix", "");
+		TransferManager tx = TransferManagerBuilder.defaultTransferManager();
+		Predicate<String> keyPredicate = getKeyFilterByFileNamePrefix(s3filePrefix);
+		KeyFilter keyFilter = s3filePrefix.isEmpty() ? KeyFilter.INCLUDE_ALL :
+			objectSummary -> keyPredicate.test(objectSummary.getKey());
+		try {
+			tx.downloadDirectory(bucket, s3prefix, new File(localFolder), keyFilter).waitForCompletion();
+		} catch (InterruptedException e) {
+			System.out.println("Transfer interrupted");
+		} finally {
+			tx.shutdownNow();
+		}
+	}
+
+	private static Predicate<String> getKeyFilterByFileNamePrefix(String s3filePrefix) {
+		if (s3filePrefix.isEmpty()) {
+			return key -> true;
+		} else {
+			return key -> {
+				String[] parts = key.split("/");
+				String fileName = parts[parts.length - 1];
+				return fileName.startsWith(s3filePrefix);
+			};
+		}
+	}
+
+	private static void deleteFile(ParameterTool params) {
+		final String bucket = params.getRequired("bucket");
+		final String s3file = params.getRequired("s3file");
+		AmazonS3ClientBuilder.defaultClient().deleteObject(bucket, s3file);
+	}
+
+	private static void deleteByPrefix(ParameterTool params) {
```

Comment:
Rename to `deleteFilesWithFilenamePrefix`.

Reply:
this also deletes by full s3 object key prefix, could be then:
`deleteByFullPrefix`